drafts/zip-user-defined-assets.rst

```diff
@@ -0,0 +1,278 @@
+asset types across all mediums would enable strong privacy at their base layer. There are
+various reasons why this is unlikely to occur in the near term, and thus it would be
+useful if these other assets could benefit from Zcash's deployment of privacy technology.
+
+Separately, there has been significant interest over the past few years in user-defined
+assets, where issuance is controlled above layer 1, but the asset is transacted on layer 1
+alongside the base asset. The most notable instance of this has been ERC-20 tokens on the
+Ethereum block chain. Adding this functionality to the Zcash network, particularly when
+coupled with privacy, enables interesting new use cases such as private polling.
+
+Both of these goals can be served by enabling multiple asset types to exist within a
+shielded pool.
+
+
+Conventions
+===========
+
+We use the following notation as defined in the Zcash protocol specification
+[#spec-notation]_, reproduced here for convenience:
+
+- *a* || *b* means the concatenation of sequences *a* then *b*.
+
+
+Specification
+=============
+
+The proposal leverages the use of homomorphic Pedersen commitments in the Sapling design.
+Value commitments for Sapling spends and outputs are balanced as follows:
+
+.. code::
+
+    cv_s1 = [v1] G_zec + [r1] H
+    cv_s2 = [v2] G_zec + [r2] H
+    cv_o1 = [v3] G_zec + [r3] H
+    cv_o1 = [v4] G_zec + [r4] H
+
+    (cv_s1 + cv_s2) - (cv_o1 + cv_o2) - [valueBalance] G_zec = bvk = [bsk] H
+
+which leads to:
+
+- ZEC balance: ``(v1 + v2) - (v3 + v4) = valueBalance``
+- bindingSig secret key: ``(r1 + r2) - (r3 + r4) = bsk``
+
+The key intuition is that if we have another value commitment value base G_asset, which
+is independent of the ZEC base G_zec, then we can balance multiple asset types
+simultaneously:
+
+.. code::
+
+    cv_s1 = [v1] G_zec + [r1] H
+    cv_s2 = [v2] G_asset + [r2] H
+    cv_o1 = [v3] G_zec + [r3] H
+    cv_o2 = [v4] G_asset + [r4] H
+    cv_o3 = [v5] G_asset + [r5] H
+
+    (cv_s1 + cv_s2) - (cv_o1 + cv_o2 + cv_o3)
+      - [valueBalance] G_zec
+      - [assetBalance] G_asset = bvk = [bsk] H
+
+which leads to:
+
+- ZEC balance: ``v1 - v3 = valueBalance``
+- BTC balance: ``v2 - (v4 + v5) = assetBalance``
+- bindingSig secret key: ``(r1 + r2) - (r3 + r4 + r5) = bsk``
+
+The circuit changes in this ZIP are specified relative to the Sapling circuit. In a
+production deployment, they would be implemented as part of a new circuit derived from
+Sapling, and would (most likely, if note commitments for ZEC are altered) form a new
+shielded pool. We refer to the new circuit and shielded pool inline below as AssetPool.
+
+Asset definitions
+-----------------
+
+An asset is defined by its asset type: a byte string that uniquely identifies the asset
+amongst all possible assets.
+
+Each asset type has an associated value commitment generator, derived as follows:
+
+.. code::
+
+    TODO: Specify GroupHash using Rescue
```

Comment:
I have an elegantly awful way to specify this so that it ends up with the same \mathcal{V} generator as Sapling does when the asset is ZEC (or TAZ on testnet).